cub/cub/device/dispatch/tuning/tuning_select_if.cuh

```diff
@@ -746,6 +861,604 @@ struct sm100_tuning<Input, flagged::yes, keep_rejects::no, offset_size::_4, prim
+                    offset_size::_8,
+                    primitive::yes,
+                    input_size::_2,
+                    may_alias::no,
+                    distinct_partitions::no>
+{
+  // trp_1.ld_0.ipt_22.tpb_288.ns_124.dcid_2.l2w_690 1.202783  1.000000  1.183737  1.311755
+  static constexpr int nominal_4b_items              = 22;
+  static constexpr int threads                       = 288;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = exponential_backoff_constructor_t<124, 690>;
+};
+
+template <class Input>
+struct sm100_tuning<Input,
+                    flagged::no,
+                    keep_rejects::yes,
+                    offset_size::_8,
+                    primitive::yes,
+                    input_size::_4,
+                    may_alias::no,
+                    distinct_partitions::no>
+{
+  // trp_1.ld_0.ipt_19.tpb_608.ns_1884.dcid_6.l2w_950 1.250302  0.988124  1.225191  1.392931
+  static constexpr int nominal_4b_items              = 19;
+  static constexpr int threads                       = 608;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = exponential_backon_jitter_constructor_t<1884, 950>;
+};
+
+template <class Input>
+struct sm100_tuning<Input,
+                    flagged::no,
+                    keep_rejects::yes,
+                    offset_size::_8,
+                    primitive::yes,
+                    input_size::_8,
+                    may_alias::no,
+                    distinct_partitions::no>
+{
+  // trp_1.ld_0.ipt_23.tpb_416.ns_0.dcid_2.l2w_1200 1.156864  1.011990  1.152368  1.266667
+  static constexpr int nominal_4b_items              = 23;
+  static constexpr int threads                       = 416;
+  static constexpr BlockLoadAlgorithm load_algorithm = BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr CacheLoadModifier load_modifier   = LOAD_DEFAULT;
+  using delay_constructor                            = exponential_backoff_constructor_t<0, 1200>;
+};
+
+#if CUB_IS_INT128_ENABLED
+// because we introduced cases for when offset is I64 this leads to regressions if not defaulted explicitly
+template <distinct_partitions DistinctPartitions>
+struct sm100_tuning<__int128_t,
+                    flagged::no,
+                    keep_rejects::yes,
+                    offset_size::_8,
+                    primitive::no,
+                    input_size::_16,
+                    may_alias::no,
+                    DistinctPartitions>
+    : sm90_tuning<__int128_t, flagged::no, keep_rejects::yes, offset_size::_4, primitive::no, input_size::_16>
+// ^^^^^ this base is wrong and leads to regressions ^^^^^
+{};
+#endif // CUB_IS_INT128_ENABLED
```

Comment:
@gonidelis I need more information on why this specialization is needed, since I cannot understand which other `sm100_tuning` could match an `input_size::_16`.

Also, what do we need to do about the
> this base is wrong and leads to regressions
?

In my understanding, we can simple remove this tuning entirely. Can you test whether works out fine?

Reply:
Seems like it slipped through during the early stages of evaluation before we fix any bugs. Looking into it rn